File: binV9/second_problem.js
const posts = {
  post_1: { tags: ["ruby", ["cli", "json"], "parsing"] },
  post_2: { tags: ["ruby", ["api", ["cli"]]] },
};

const tags = Object.values(posts).flatMap((post) => {
  return post.tags.flat(Infinity);
});

const uniqTags = [...new Set(tags)];

console.log(JSON.stringify({ data: uniqTags }, null, 2));

File: binsV1/binV7/second_problem.rb
require 'json'

posts = {
  "post_1": { "tags": [ "ruby", [ "cli", "json" ], "parsing" ] },
  "post_2": { "tags": [ "ruby", [ "api", [ "cli" ] ] ] }
}

tags = posts.values.flat_map do |post|
  post[:tags].flatten
end.uniq

print JSON.pretty_generate({ data: tags })
